src/main/java/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -308,13 +309,12 @@ private void createAndSaveStats(final boolean showPhaseStats) {
       text.append("\n");
       clone.getHistory().gotoNode(clone.getHistory().getLastNode());
       @SuppressWarnings("unchecked")
-      final Enumeration<HistoryNode> nodes =
-          ((DefaultMutableTreeNode) clone.getHistory().getRoot()).preorderEnumeration();
+      final Enumeration<TreeNode> nodes = ((DefaultMutableTreeNode) clone.getHistory().getRoot()).preorderEnumeration();
```

Comment:
It might be cleaner to encapsulate all the casts and unchecked warnings in the `History` class:

```java
public Enumeration<HistoryNode> getNodes() {
  @SuppressWarnings("unchecked")
  final Enumeration<TreeNode> nodes = ((DefaultMutableTreeNode) getRoot()).preorderEnumeration();
  return Collections.enumeration(Collections.list(nodes).stream()
      .map(HistoryNode.class::cast)
      .collect(Collectors.toList()));
}
```

The problem with the above version is that it eagerly converts the enumeration to a list.  If the number of tree nodes is fairly large, the following solution avoids the intermediate list at the cost of some complexity:

```java
public Enumeration<HistoryNode> getNodes() {
  @SuppressWarnings("unchecked")
  final Enumeration<TreeNode> nodes = ((DefaultMutableTreeNode) getRoot()).preorderEnumeration();
  final Iterator<TreeNode> iterator = Iterators.forEnumeration(nodes);
  return Iterators.asEnumeration(
      StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false)
        .map(HistoryNode.class::cast)
        .iterator());
}
```